dashboard/models.py

```diff
@@ -197,13 +199,18 @@ def json(self):
         :rtype: Dict
         """
 
-        return {
-            self.course.canvas_id: {
-                'fa': int(self.show_files_accessed),
-                'ap': int(self.show_assignment_planning),
-                'gd': int(self.show_grade_distribution),
+        try:
+            return {
+                self.course.canvas_id: {
+                    'fa': int(self.show_files_accessed),
+                    'ap': int(self.show_assignment_planning),
+                    'gd': int(self.show_grade_distribution),
+                }
             }
-        }
+        except ObjectDoesNotExist:
```

Comment:
```suggestion
        except (ObjectDoesNotExist, Exception) as e:
```

Reply:
Sometimes we get exceptions that are unknown and  I would like to also capture the generic case

Reply:
Yeah, I could catch more here, though it's looks like the best patterns are to except the specific exceptions separate from the less specific ones.

https://docs.quantifiedcode.com/python-anti-patterns/correctness/bad_except_clauses_order.html

Catching the general Exception case usually isn't good either as it can potentially hide real issues that aren't noticed. I think for this feed we never want an error, maybe the general Exception should be moved up into the view instead.